src/Temporalio/Bridge/src/metric.rs

```diff
@@ -455,11 +510,50 @@ impl metrics::Histogram for CustomMetricInteger {
     }
 }
 
-impl metrics::Gauge for CustomMetricInteger {
+impl metrics::HistogramF64 for CustomMetric {
+    fn record(&self, value: f64, attributes: &metrics::MetricAttributes) {
+        unsafe {
+            let meter = &*(self.meter_impl.0);
+            (meter.metric_record_float)(
+                self.metric,
+                value,
+                raw_custom_metric_attributes(attributes),
+            );
+        }
+    }
+}
+
+impl metrics::HistogramDuration for CustomMetric {
+    fn record(&self, value: Duration, attributes: &metrics::MetricAttributes) {
+        unsafe {
+            let meter = &*(self.meter_impl.0);
+            (meter.metric_record_duration)(
+                self.metric,
+                value.as_millis().try_into().unwrap_or(u64::MAX),
```

Comment:
I can't decide in the rare case that this happens if it's better to record this max or simply not record anything at all

Reply:
I was stuck a bit here too and just settled on this. Hopefully the issue never arises. But if we think it's enough of an issue, we can look info logging instead.

This gets even worse when we convert back to .NET types because. E.g. for .NET a `ulong` (i.e. `u64`) is "not CLS-compliant", so it's best to use `long` (i.e. `i64`) instead, so I limit to max there too (default unchecked cast wraps into negative).